### ISSUE TITLE
[Feature] Attribute selector injection allowed to be set to optional.

### ISSUE THREAD

**Issue by yb199478** [CatLib/Core#253]
Attribute selector injection allowed to be set to optional.

use to open this feature:

```csharp
[Inject(Required = false)]
```

This is useful for some property selectors with initial defaults:

```csharp
[Inject(Required = false)]
public string Name { get; set; } = "foo";
```

**Comment by yb199478** [comment]
resolved #253

**Comment by yb199478** [comment]
Resolved in #255
